### DISCUSSION TITLE
Geoinformation

### DISCUSSION THREAD

**Discussion by m-appel** [InternetHealthReport/internet-yellow-pages#158] [Ideas]
Collecting different sources for geoinformation here that could be helpful with modelling:

- [Geonames](https://download.geonames.org/export/dump/)
- [Geofeed](https://geolocatemuch.com/)

Geofeed only contains Country and City name, which might break thinks in the US where it is not uncommon to have multiple cities with the same name in different states.

In general normalization (regions, city names, etc.) between different datasets might be problematic.

**Comment by romain-fontugne** [comment]
For Geonames we'll have geo-coordinates (lat,lon) that we need to model. 

I suggest to start modelling the coordinates that are already present in the database. I found coordinates related to `AtlasProbe`, `AS` (data from ASRank), `Facility` and `IXP` from PeeringDB.

For these entities, we can add a relationship `LOCATED_AT` to a new node type `Point` that gives their geographical position. 
Something like this:
```
(:AtlasProbe)-[:LOCATED_AT]->(:Point {position: point({longitude: 12.3, latitude: 45.6})
```

Then, in a second step we can add geonames cities with their coordinates. This would allow to do most of the basic searches, e.g. find all resources around one city. 

Finally, if we want to map better Points to their regions/cities we can import geonames admin areas and corresponding polygons, and do a one-to-one mapping in a post-script.

**Comment by Vrittigyl** [comment]
The Point-based modelling you described can be implemented for the entities that already have coordinates.

Specifically, the existing crawlers can be extended to normalise coordinates into Point nodes with LOCATED_AT relationships for:
- AS (from ASRank)
- IXP (from PeeringDB)
- Facility (from PeeringDB)

This would use the same pattern already established for AtlasProbe nodes.
I'm happy to work on this if you'd like to move forward with it before tackling the geonames integration.

**Reply by romain-fontugne** [reply]
sorry the discussion is a bit old and we haven't closed/updated it. Location for AS, IXP, and Facility is now implemented.

**Reply by Vrittigyl** [reply]
Thanks for the update!
